### PR TITLE
Add table of contents to Downloads Page

### LINKED ISSUE #1020: Figure out Production/Research dichotomy on Downloads Page

### PR DESCRIPTION
**Pull request by rileyhgrant**
Resolves #1020 

---

**Background**:
As part of discussing the Research vs Production datasets, several changes were requested. The current format of the downloads page is large and unweildy, users who know what they are looking for cannot `ctrl + f` to find it, due to the tabs, the datasets themselves are huge and easy to be lost in, and there is a sub optimal way to differentiate between production dataset and research/supplemental datasets, as each individual Download category (v2, v3, etc) should have a production / research split.

---

Changes the terminology of Production / Research downloads to Core Dataset and Secondary Analyses, per slack and meeting discussions.

Adds a table of contents to the Downloads Page, per gnomAD meeting. This allows for all of the downloads to be present on the page at one time if a user wishes to use `ctrl + f` and additionally allows for a user to know their location the page at any given time, and know what other datasets exist at a glance. The indentation lets their be subcategories within any given large category (v2, v3, etc). 

The table of contents allows for users to quickly navigate to a selection by clicking it, and highlights the section a user is in. Due to the size of the available downloads, only the individual download links for the given downloads category appear on the table of contents.

---

Here, only the top level headers show since the user is on the summary/intro section
Screenshot 2023-01-26 at 16 02 30

Since the user is on the v2 variants section, all of the links for v2 show in the table of contents
Screenshot 2023-01-26 at 16 02 47

Here, since the user is in the v3 section, those links appear.
Screenshot 2023-01-26 at 16 02 52

### CODE REVIEW

**Comment by rileyhgrant** [comment]
The first commit is the main addition of the Table of Contents for the Downloads page. The second commit adds a couple additional dependencies to allow loading css modules, as I could not find another clean way to only add `scroll-behavior: smooth;` to only the Downloads page.

I figure we can decide whether or not the scroll behavior is worth the dependencies / if there is a way to make it work with Styled Components for only the Downloads page.

File: browser/src/DownloadsPage/DownloadsPage.module.css [hunk 1]
```diff
@@ -0,0 +1,8 @@
+/*
+	This CSS module is used alongside Styled Components to allow modification of a single
+	html page for smooth scrolling
+*/
+
+html {
+  scroll-behavior: smooth;
```
Comment (mattsolo1) [inline browser/src/DownloadsPage/DownloadsPage.module.css:7]: I tried this with smooth and with auto.

Smooth scrolling is a cool effect! It gives a bit of context to how far up or down I'm jumping on the page.

On the other hand It felt a bit janky when jumping from the very top the very bottom. Not sure how to describe it. Without smooth scroll, it felt much faster to jump to different datasets. I know that's boring though!

Could be my personal preference though... I'd suggest seeking feedback from others.

**Review by mattsolo1 — APPROVED** [review]
I love the redesign! Just see my comments re: smooth scroll. I haven't tried any of the links -- just going to assume everything working as before.

Great stuff.

**Comment by rileyhgrant** [comment]
Ok, I was foolishly waiting on this other dataset to come in to be added, but you are totally right Matt that I might as well just merge this and push this to live in the meantime.

Initially, it seemed like this downloads PR would be more urgent, as it was blocking the addition of the other dataset. 

However, since it has now been over a week, it seems as though there were other things blocking that dataset. As such, this PR will be merged and pushed to prod, and when the new dataset comes in (I believe it's a subset of HGDP + 1KG as part of tutorials made by Alicia's group) there can be a new PR/commit for that.

**Comment by rileyhgrant** [comment]
Comments from Slack discussion have been addressed, some rewording of content and making clearer the hierarchy on the page by way of different font size for different level section headers

Screenshot 2023-02-15 at 08 50 59

**Pull request merged**